### ISSUE TITLE
Zoom To Center of Image

### ISSUE THREAD

**Issue by shawnbro** [prc5/react-zoom-pan-pinch#86]
I am trying to attach a slider component to trigger zoom in & out, I use it to update the scale value however when that happens it zooms into (0,0) (top, left) corner of the image. Is there an easy way to have this automatically zoom to the center of the image?

**Comment by kylegoines** [comment]
Hi there, im going out on a limb, as im quite new to the libairy myself, but i believe you should be able to set a center point with `setPositionX ` and `setPositionY`
these methods return from the TransformWrapper. Similar to `zoomIn`, and `ZoomOut` from the example

**Comment by chienweiluo** [comment]
Hi @kylegoines, i have met this issue, too. Could you please share your workaround here? 

I use `setPositionX ` and  `setPositionY ` before `setScale` but it seems not working and the transform point still `(0, 0)`.

Many thanks.

```
onClick={() => {
                            setPositionX(500)
                            setPositionY(100)

                            setScale(1)
                        }}
```

**Comment by petya-v** [comment]
Hi, sorry I'm really struggling here. 
I noticed that the zoom buttons already have this functionality. However they use a bunch of stuff that doesn't seem to be exposed.

This is as far as I got, from: https://github.com/prc5/react-zoom-pan-pinch/blob/f78c546414b13c61a68368db74ed9fd9fb2e185c/src/store/zoom/index.ts#L268

```
            const wrapperBounds = wrapperRef?.current?.getBoundingClientRect();
            resetTransform();
            const x = (wrapperBounds.width / 2) / defaultScale;
            const y = (wrapperBounds.height / 2) / defaultScale;
            setTransform(x, y, defaultScale);
```

Is anyone able to tell me how I can zoom to the center?
Seems like everyone on github (including the suggestion from @kylegoines) seems to be hardcoding the center point, which doesn't work for my case as I'm being passed a dynamic image:
https://github.com/search?l=JavaScript&q=react-zoom-pan-pinch+center&type=Code
